extensions/markdown-language-features/src/languageFeatures/copyFiles/copyPasteLinks.ts

```diff
@@ -43,6 +40,86 @@ class PasteLinkEditProvider implements vscode.DocumentPasteEditProvider {
 	}
 }
 
+// function checkURL(item: vscode.DataTransferItem | undefined) {
+// 	try {
+// 		const url = new URL(item?.value);
+// 		return externalUriSchemes.includes(url.protocol.slice(0, -1));
+// 	} catch (error) {
+// 		return false;
+// 	}
+// }
+
+const CONTROL_CODES = '\\u0000-\\u0020\\u007f-\\u009f';
+const WEB_LINK_REGEX = new RegExp('(?:[a-zA-Z][a-zA-Z0-9+.-]{2,}:\\/\\/|data:|www\\.)[^\\s' + CONTROL_CODES + '"]{2,}[^\\s' + CONTROL_CODES + '"\')}\\],:;.!?]', 'ug');
+const WIN_ABSOLUTE_PATH = /(?<=^|\s)(?:[a-zA-Z]:(?:(?:\\|\/)[\w\.-]*)+)/;
+const WIN_RELATIVE_PATH = /(?<=^|\s)(?:(?:\~|\.)(?:(?:\\|\/)[\w\.-]*)+)/;
+const WIN_PATH = new RegExp(`(${WIN_ABSOLUTE_PATH.source}|${WIN_RELATIVE_PATH.source})`);
+const POSIX_PATH = /(?<=^|\s)((?:\~|\.)?(?:\/[\w\.-]*)+)/;
+const LINE_COLUMN = /(?:\:([\d]+))?(?:\:([\d]+))?/;
+const isWindows = (typeof navigator !== 'undefined') ? navigator.userAgent && navigator.userAgent.indexOf('Windows') >= 0 : false;
+const PATH_LINK_REGEX = new RegExp(`${isWindows ? WIN_PATH.source : POSIX_PATH.source}${LINE_COLUMN.source}`, 'g');
+
+const MAX_LENGTH = 2000;
+
+type LinkKind = 'web' | 'path' | 'text';
+type LinkPart = {
+	kind: LinkKind;
+	value: string;
+};
+
+function detectLinks(text: string): boolean | undefined {
```

Comment:
Instead of implementing this yourself, can you try seeing if `Uri.parse` works? You can also write a few tests for this as this can also help guide the implementation

If `Uri.parse` or `new Url` works, then you also might need some extra validation, such as checking that the input string does not contain anywhere space for instance